contracts/AllDay.cdc

```diff
@@ -611,15 +623,21 @@ pub contract AllDay: NonFungibleToken {
         // borrowNFT gets a reference to an NFT in the collection
         //
         pub fun borrowNFT(id: UInt64): &NonFungibleToken.NFT {
-            return &self.ownedNFTs[id] as &NonFungibleToken.NFT
+            pre {
+                self.ownedNFTs[id] != nil: "Cannot borrow NFT, no such id"
+            }
+
+            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
```

Comment:
Used the force unwrap here because we have already nil checked in the precondition and I figured it would be easier than changing the function signature

Reply:
(Open to other suggestions here though!)

Reply:
yeah, I think in an ideal world, we would return an optional, but that would require changing the standard, which would be a breaking change unfortunately